src/proj_functions.jl

```diff
@@ -0,0 +1,211 @@
+
+function transform!(src::Projection, dest::Projection, position::Array{Float64,2}, radians::Bool=false)
+    ndim = size(position,2)
+    ndim == 2 && return transform2!(src, dest, position, radians)
+    ndim == 3 && return transform3!(src, dest, position, radians)
+    error("position must be Nx2 or Nx3")
+end
+transform!(src::Projection, dest::Projection, position::Array{Float64,2}, radians::Bool=false) =
+    transform!(src, dest, copy(position), radians)
+
+function transform!(src::Projection, dest::Projection, position::Vector{Float64}, radians::Bool=false)
+    ndim = length(position)
+    ndim == 2 && return transform2!(src, dest, position, radians)
+    ndim == 3 && return transform3!(src, dest, position, radians)
+    error("position must be 2 or 3-dimensional")
+end
+transform(src::Projection, dest::Projection, position::Vector{Float64}, radians::Bool=false) =
+    transform!(src, dest, copy(position), radians)
+
+# Unused/untested
+# @doc """
+# Return the lat/long coordinate system on which a projection is based.
+# If the coordinate system passed in is latlong, a clone of the same will be returned.
+# """ ->
+# latlong_projection(proj::Projection) = Projection(_latlong_from_proj(proj.rep))
+
+@doc """
+Solve the direct geodesic problem.
+
+Args:
+
+    position - coordinates of starting location, modified in-place to [dest] (described below)
+    azimuth  - azimuth (degrees) ∈ [-540, 540)
+    distance - distance (metres) to move from (lat,lon); can be negative
+    proj     - the given projection whose ellipsoid we move along
+   
+Returns:
+
+    dest     - destination after moving for [distance] metres in [azimuth] direction.
+    azi      - forward azimuth (degrees) at destination [dest].
+""" ->
+function geod_direct!(position::Vector{Float64}, azimuth::Float64, distance::Float64, proj::Projection)
+    xy2lonlat!(position, proj)
+    dest, azi = _geod_direct!(proj.geod, position, azimuth, distance)
+    lonlat2xy!(dest, proj), azi
+end
+
+@doc """
+Solve the direct geodesic problem.
+
+Args:
+
+    lonlat   - latitude, longitude (degrees) ∈ [-90, 90]
+    azimuth  - azimuth (degrees) ∈ [-540, 540)
+    distance - distance (metres) to move from (lat,lon); can be negative
+    proj     - the given projection whose ellipsoid we move along
+   
+Returns:
+
+    dest     - destination after moving for [distance] metres in [azimuth] direction.
+    azi      - forward azimuth (degrees) at destination [dest].
+
+""" ->
+geod_direct(position::Vector{Float64}, azimuth::Float64, distance::Float64, proj::Projection) = 
+    geod_direct!(copy(position), azimuth, distance, proj)
+
+@doc "Returns the destination by moving along the ellipsoid in the given projection" ->
+destination!(position::Vector{Float64}, azi::Float64, dist::Float64, proj::Projection) = geod_direct!(position, azi, dist, proj)[1]
+destination(position::Vector{Float64}, azi::Float64, dist::Float64, proj::Projection) = destination!(copy(position), azi, dist, proj)
```

Comment:
In the turf example, it appears the usage would be namespaced as `turf.destination` - that's not hard to understand when reading code which uses turf.  For exported functions in julia it's less common to be explicitly writing namespaces, so IMO generic names should apply only to similarly generic functionality.